boofuzz/monitors/imonitor.py

```diff
@@ -0,0 +1,123 @@
+import abc
+
+from future.utils import with_metaclass
+
+
+class IMonitor(with_metaclass(abc.ABCMeta, object)):
```

Comment:
Now having a little distance due being on vacation, I don't recall exactly why I went this route. I guess I wanted to implement reasonable defaults for some of the functions in case a Monitor doesn't support them, but then again I don't have an explanation why restart_target is abstract - or retrieve_data is, and get_crash_synopsis isn't (I guess my reasoning there was: every monitor may provide data, but only those who can detect crashes have a crash synopsis? something like this).

I'm open to suggestions on how to proceed. In doubt, I'd say we remove the default implementations and move everything to the child classes.

Reply:
I'd say all one way or all the other. I like the idea of being able to implement only one method to keep implementation classes cleaner, and also to keep you from needing to update all implementations if a method is added.

Reply:
I've renamed the class to BaseMonitor and removed the metaclass. Hopefully this also cleans up stickler. All methods have default implementations now with a line in the docs stating what the default is.

Should be good now.